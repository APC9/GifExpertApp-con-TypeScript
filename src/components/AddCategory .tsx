import { ChangeEvent, FormEvent, useState } from 'react';

interface Props {
  onNewCategory: (newCategory: string) => void;
}

export const AddCategory = ({ onNewCategory }: Props) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = (event: ChangeEvent<HTMLInputElement>) => {
    const input = event.target.value;
    setInputValue(input);
  };

  const onSubmit = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    if (inputValue.trim().length <= 1) return;

    //setCategories(categories => [inputValue, ...categories])

    onNewCategory(inputValue.trim());
    setInputValue('');
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};
