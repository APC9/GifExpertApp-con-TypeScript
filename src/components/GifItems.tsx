interface Props {
  title: string;
  url: string;
}

export const GifItems = ({ title, url }: Props) => {
  return (
    <div className="card-grid card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};
