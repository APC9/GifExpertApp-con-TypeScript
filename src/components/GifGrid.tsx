import { GifItems } from './GifItems';
import { useFetchGifs } from '../hook/useFetchGifs';

interface Props {
  category: string;
}

export const GifGrid = ({ category }: Props) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Cargando...</h2>}

      <div className="card-grid">
        {images?.map(({ id, title, images }) => (
          <GifItems key={id} title={title} url={images.downsized_medium.url} />
        ))}
      </div>
    </>
  );
};
