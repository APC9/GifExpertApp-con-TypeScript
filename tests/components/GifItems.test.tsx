import React from 'react';
import { render, screen } from '@testing-library/react';
import { GifItems } from '../../src/components/GifItems';

describe('Test <GifItems /> ', () => {
  const title = 'Goku';
  const url = 'https://media.giphy.com/media/l0000';

  test('debe de hacer match con el snapchot', () => {
    const { container } = render(<GifItems title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar la imagen con el url y el Alt indicado', () => {
    render(<GifItems title={title} url={url} />);

    //expect(screen.getByRole<HTMLImageElement>('img').src).toBe(url);
    const { src, alt } = screen.getByRole<HTMLImageElement>('img');
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test('Debe de monstrar el titulo en el componente', () => {
    render(<GifItems title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
