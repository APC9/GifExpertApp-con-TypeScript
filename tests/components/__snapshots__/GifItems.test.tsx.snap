// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifItems />  debe de hacer match con el snapchot 1`] = `
<div>
  <div
    class="card-grid card"
  >
    <img
      alt="Goku"
      src="https://media.giphy.com/media/l0000"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
